clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -18,9 +18,22 @@
 #include "clang/CIR/MissingFeatures.h"
 #include "llvm/IR/IntrinsicsX86.h"
 
+#define UNIMPLEMENTED_BUILTIN()                                                \
+  do {                                                                         \
+    cgm.errorNYI(e->getSourceRange(),                                          \
+                 std::string("unimplemented X86 builtin call: ") +             \
+                     getContext().BuiltinInfo.getName(builtinID));             \
+    return {};                                                                 \
+  } while (0)
+
 using namespace clang;
 using namespace clang::CIRGen;
 
+/// Get integer from a mlir::Value that is an int constant or a constant op.
+static int64_t getIntValueFromConstOp(mlir::Value val) {
+  return val.getDefiningOp<cir::ConstantOp>().getIntValue().getSExtValue();
+}
```

Comment:
In the incubator we define this static helper in both `clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp` and `clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp`, and we also frequently use the same basic pattern with `getZExtValue()` in the X86 code. Maybe it would be better to move this to `CIRGenFunction`.

Also, there should be some validation that the argument really is a ConstantOp.

```suggestion
int64_t getIntValueFromConstOp(mlir::Value val) {
  auto constOp = val.getDefiningOp<cir::ConstantOp>();
  assert(constOp && "getIntValueFromConstOp call with non ConstantOp");
  return constOp.getSExtValue();
}
```